File: JavaScript-Solutions/check-for-palindrome.js
function palindrome(str) {
  var toRemove = /[.,_:*+?^${}#@%$&()|[\]\\\-]|\s/g;
  var cleanedStr = str.toLowerCase().replace(toRemove, "");
  var reversedStr = cleanedStr.split('').reverse().join('');
  return reversedStr === cleanedStr;
}

File: Java-Solutions/check-for-palindrome.java
    public static boolean checkForPalindromes (String str) {
        String toRemove = "[!@#$%^&*:><?;\\}\\{\\s]";
        String cleanedStr = str.toLowerCase().replaceAll(toRemove, "");
        String reversedStr = new StringBuilder(cleanedStr).reverse().toString();
        return cleanedStr.equals(reversedStr);
    }
